### PR TITLE
[Artiom Habarov] - Added home work (Primitives)

### PR DESCRIPTION
**Pull request by ahabarov**

### CODE REVIEW

File: Artiom/src/main/java/Circle_Area.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+public class Circle_Area {
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Circle_Area.java]: Don't use underscore in class's name
Naming convention: Class names should be nouns with the first letter of each internal word capitalized.

File: Artiom/src/main/java/Circle_Area.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    public static void main(String[] args) {
+        int radius = 4;
+        double pi = 3.14159265359;
+        double Area = pi * (radius * radius);
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Circle_Area.java]: Please follow the naming convention (local variable name - first letter lowercase)

File: Artiom/src/main/java/Primitives.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+//3. Write a java program to compare two strings lexicographically, ignoring case differences, if strings are different concatenate a given string to the end of another string.
+
+public class Primitives {
+
+    public static void main(String[] args) {
+        String lower_string = "string";
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Primitives.java]: Please follow the naming convention

File: Artiom/src/main/java/Circle_Area.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+public class Circle_Area {
+    public static void main(String[] args) {
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Circle_Area.java]: It's a good practice to move logic to the separate method and use main method for high level operations.
So here you can create a separate method which will accept one parameter (radius) and return the area.
Use this method from main to print calculated area.

File: Artiom/src/main/java/DateTemplate.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+public class DateTemplate {
+
+    public static void main(String[] args) {
+        String dateRegex = "2020-09-03";
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/DateTemplate.java]: Looks fine but lets make it pass the following tests as well:
0000-00-00 - is not a valid date
0000-19-00 - is not a valid date
0000-00-39 - is not a valid date
2020a03b08 - is not a valid date

File: Artiom/src/main/java/Months.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            case 11:
+                monthName = "November";
+                break;
+            case 12:
+                monthName = "December";
+                break;
+            default:
+                monthName = "Invalid month";
+                break;
+
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Months.java]: Please remove empty line

File: Artiom/src/main/java/Primitives.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+        String upper_String = "String";
+        if (lower_string != upper_String) {
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Primitives.java]: Its not save to use == operator to compare strings
see also https://www.programiz.com/java-programming/examples/compare-strings#:~:text=You%20should%20not%20use%20%3D%3D,and%20not%20the%20object%20itself.

**Comment by venache1** [comment]
Please squash your commits after fixing all messages.

File: Artiom/src/main/java/Circle_Area.java (hunk 1 above)
Comment (ahabarov) [reply Artiom/src/main/java/Circle_Area.java]: Done

File: Artiom/src/main/java/Circle_Area.java (hunk 2 above)
Comment (ahabarov) [reply Artiom/src/main/java/Circle_Area.java]: done

File: Artiom/src/main/java/Primitives.java (hunk 3 above)
Comment (ahabarov) [reply Artiom/src/main/java/Primitives.java]: done

File: Artiom/src/main/java/DateTemplate.java (hunk 5 above)
Comment (ahabarov) [reply Artiom/src/main/java/DateTemplate.java]: done

File: Artiom/src/main/java/Months.java (hunk 6 above)
Comment (ahabarov) [reply Artiom/src/main/java/Months.java]: done

File: Artiom/src/main/java/Primitives.java (hunk 7 above)
Comment (ahabarov) [reply Artiom/src/main/java/Primitives.java]: done

File: Artiom/src/main/java/Circle_Area.java (hunk 4 above)
Comment (ahabarov) [reply Artiom/src/main/java/Circle_Area.java]: done

File: Artiom/src/main/java/Primitives.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+//3. Write a java program to compare two strings lexicographically, ignoring case differences, if strings are different concatenate a given string to the end of another string.
+
+public class Primitives {
+
+    public static void main(String[] args) {
+        String lower_string = "spring";
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Primitives.java]: Could you please rename your variables according to the naming convention

File: Artiom/src/main/java/CircleArea.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+public class CircleArea {
+    public static void main(String[] args) {
+        //calculateCircleArea();
```
Comment (venache1) [inline Artiom/src/main/java/CircleArea.java]: Remove comented code

File: Artiom/src/main/java/Primitives.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        String upper_String = "Spring";
```
Comment (venache1) [inline Artiom/src/main/java/Primitives.java]: Please rename those variables acording Java naming convention
https://www.javatpoint.com/java-naming-conventions

File: Artiom/src/main/java/CircleArea.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+        System.out.println(calculateCircleArea());
+    }
+    public static double calculateCircleArea() {
+
+        int radius = 4;
+        double pi = 3.14159265359;
+        double area = pi * (radius * radius);
+
```
Comment (venache1) [inline Artiom/src/main/java/CircleArea.java]: Remove empty line

File: Artiom/src/main/java/CircleArea.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+public class CircleArea {
+    public static void main(String[] args) {
+        //calculateCircleArea();
+        System.out.println(calculateCircleArea());
+    }
+    public static double calculateCircleArea() {
+
```
Comment (venache1) [inline Artiom/src/main/java/CircleArea.java]: Remove empty line

File: Artiom/src/main/java/Numbers.java [hunk 13]
```diff
@@ -0,0 +1,10 @@
+//Display numbers from 1 to 10 using for()
+
+public class Numbers {
+
+    public static void main(String[] args) {
+
```
Comment (venache1) [inline Artiom/src/main/java/Numbers.java]: Remove empty line

File: Artiom/src/main/java/Primitives.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+        if (lower_string.equals(upper_String))
+            System.out.println("Strings are not different");
+        else
+            System.out.println(lower_string + upper_String);
+    }
+
```
Comment (venache1) [inline Artiom/src/main/java/Primitives.java]: Remove empty line

File: Artiom/src/main/java/Primitives.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+    }
+
+
```
Comment (venache1) [inline Artiom/src/main/java/Primitives.java:13]: Remove empty line

**Comment by venache1** [comment]
Please squash your commits

File: Artiom/src/main/java/DateTemplate.java (hunk 5 above)
Comment (VitalinaPetrenco) [reply Artiom/src/main/java/DateTemplate.java]: Your version is still not cover above test cases.
As an example you can try the next one regex:

^([1-9]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01]))$

year:
[1-9] - can match the number 1 to 9
\d{3} - searches for three digits

month:
0[1-9] - starts from 0 and following digit between 1 and 9 (month from 01 to 09)  
| - OR 
1[0-2] - starts from 1 and following digit between 0 and 2 (month: 10, 11, 12)

day: 
0[1-9] - starts from 0 and following digit between 1 and 9
|- OR 
[12]\\d - digit 1 or 2 with following any digit (\\d - short for [0-9])
|- OR 
3[01] -  starts from 3 and following digit 0 or 1

File: Artiom/src/main/java/Expeliarmus.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+public class Expeliarmus {
```
Comment (VitalinaPetrenco) [inline Artiom/src/main/java/Expeliarmus.java]: Is it "Primitives" task?
Could you please remove not Primitive tasks from this PR

File: Artiom/src/main/java/DateTemplate.java (hunk 5 above)
Comment (ahabarov) [reply Artiom/src/main/java/DateTemplate.java]: done

File: Artiom/src/main/java/Primitives.java (hunk 8 above)
Comment (ahabarov) [reply Artiom/src/main/java/Primitives.java]: done

File: Artiom/src/main/java/CircleArea.java (hunk 9 above)
Comment (ahabarov) [reply Artiom/src/main/java/CircleArea.java]: done

File: Artiom/src/main/java/Primitives.java (hunk 10 above)
Comment (ahabarov) [reply Artiom/src/main/java/Primitives.java]: done

File: Artiom/src/main/java/CircleArea.java (hunk 12 above)
Comment (ahabarov) [reply Artiom/src/main/java/CircleArea.java]: done

File: Artiom/src/main/java/Numbers.java (hunk 13 above)
Comment (ahabarov) [reply Artiom/src/main/java/Numbers.java]: done

File: Artiom/src/main/java/Primitives.java (hunk 14 above)
Comment (ahabarov) [reply Artiom/src/main/java/Primitives.java]: done

File: Artiom/src/main/java/Primitives.java (hunk 15 above)
Comment (ahabarov) [reply Artiom/src/main/java/Primitives.java:13]: done

File: Artiom/src/main/java/Expeliarmus.java (hunk 16 above)
Comment (ahabarov) [reply Artiom/src/main/java/Expeliarmus.java]: done

**Review by VitalinaPetrenco — APPROVED** [review]

**Review by venache1 — APPROVED** [review]

**Review by Claudia-20 — APPROVED** [review]

**Pull request merged**